### PR TITLE
Fix Publishing Of Website

### PR DESCRIPTION
**Pull request by adamgfraser**
It looks like the website is failing to publish because `unidoc` fails to compile. It appears there are some other issues where there can be spurious compilation fails with `unidoc`. I added type annotations where necessary to resolve the failures. Obviously these are ugly but they don't impact the user API so I think it is best to get this in so we can get the website publishing and start adding content to it.

With this change you can run the following commands to create the site:

```
sbt docs/docusaurusCreateSite
cd website
yarn sta
```

### CODE REVIEW

**Review by sideeffffect — APPROVED** [review]

**Comment by sideeffffect** [comment]
Btw, we're doing it the same way (with `unidoc` & co.) as it's done in ZIO, ZIO Query, etc, right?

**Comment by adamgfraser** [comment]
@sideeffffect Yes. I think the issue is we are doing more "fancy stuff" than those libraries are. All of the issues are with type inference for partially applied types which doesn't really come up in those libraries do to the avoidance of higher kinded types.

**Comment by sideeffffect** [comment]
Shouldn't we also add the step that generates the website to `ci/circleci: mdoc`, so that we don't introduce a regression in a future PR?

**Comment by adamgfraser** [comment]
@sideeffffect Good idea. There is a `microsite` action in the `config.yml` file that is run in ZIO Core but was commented out here that does that. I will uncomment that and then once we get that working we can add it to the branch protection rules.

**Review by sideeffffect — APPROVED** [review]

**Comment by sideeffffect** [comment]
Great job, thanks!

File: .circleci/config.yml [hunk 1]
```diff
@@ -169,33 +169,33 @@ release: &release
...
-#             node -v
-#             ./sbt ++${SCALA_VERSION}! docs/docusaurusCreateSite
-#             ./sbt ++${SCALA_VERSION}! docs/docusaurusPublishGhpages
-#       - <<: *clean_cache
-#       - <<: *save_cache
+microsite: &microsite
+  steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "11:2c:fa:98:69:d8:a1:d5:76:2e:38:d0:91:39:85:ce"
```
Comment (mijicd) [inline .circleci/config.yml:176]: Please verify this is the fingerprint set for prelude.

**Review by mijicd — COMMENTED** [review]
One small question, other than that looks good to me!

**Review by mijicd — APPROVED** [review]

**Review by sideeffffect — APPROVED** [review]

**Comment by sideeffffect** [comment]
Another thing that comes to mind is, shouldn't we also add a new CircleCI job dedicated only to _building_ the site (so only `docs/docusaurusCreateSite`), and this job will be a prerequisite of `release` job and required by GitHub to pass for merging PRs?
Should we do it in this PR, or should we merge it right now and I will open a new PR for that?

**Comment by mijicd** [comment]
@sideeffffect please let's keep the CI layout unified across the projects.

**Comment by sideeffffect** [comment]
@mijicd alright :ok_hand: 
Do you think ZIO projects in general could benefit from this additional `docs/docusaurusCreateSite` job? It seems we could accidentally merge a PR that will fail building the website, which would be unfortunate. Should we discuss this in a separate ticket/PR (since that would be more general discussion, not related to this specific PR)?

**Comment by mijicd** [comment]
Let's move it to #devops-lair on discord, as this is a bit broader topic :).

**Comment by sideeffffect** [comment]
Is there anything that blocks us from merging this PR?

**Pull request merged**